src/core_plugins/kibana/public/dashboard/dashboard_config.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Part of the exposed plugin API - do not remove without careful consideration.
+ * @type {boolean}
+ */
+let hideDashboardWriteControls = false;
```

Comment:
We should probably do this as an angular provider that exposes the ability to modify the config as a part of angular's config phase, for a few reasons:
 - values can be wrapped up in an angular app/injector instance and not be stored in module scope
 - the config can be read-only once the app starts by only exposing an api to set the value in the config phase (we are unlikely to consume this value in a way that would support runtime modification)
 - values will be set before any application code runs to prevent scenarios where some code reads the config before the intended value is available

To do this, modify this to create an angular provider that exposes the `turnHideWriteControlsOn()` method, and then an instance that exposes the `getHideWriteControls()` method:
```js
uiModules.get('kibana')
  .provider('dashboardConfig', () => {
    let hideWriteControls = false

    return {
      turnHideWriteControlsOn() {
        hideWriteControls = true
      },
      $get() {
        return {
          getHideWriteControls() {
            return hideWriteControls
          }
        }
      }
    };
  })
```

Then people can change the config value within angular's config phase:
```js
uiModules.get('kibana')
  .config(dashboardConfigProvider => dashboardConfigProvider.turnHideWriteControlsOn())
```

And dashboard can access the config value by injecting the `dashboardConfig`:
```js
export function DashboardListingController(..., dashboardConfig) {
  // ...
  this.hideWriteControls = dashboardConfig.getHideWriteControls()
  // ... 
}
```